recipes/peg/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+from conans import ConanFile, tools, MSBuild
+import os
+
+class PegConan(ConanFile):
+    name = "peg"
+    license = "MIT"
+    description = "peg and leg are tools for generating recursive-descent parsers"
+    topics = ("conan", "peg", "leg", "parser")
+    homepage = "https://www.piumarta.com/software/peg/"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _source_subfolder(self):
+        return "bin"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        if self.settings.os == "Windows":
+            with tools.chdir(self._source_subfolder):
+                with tools.vcvars(self.settings):
+                    msbuild = MSBuild(self)
+                    if self.settings.arch == "x86":
+                        msbuild.build_env.link_flags.append("/MACHINE:X86")
+                    elif self.settings.arch == "x86_64":
+                        msbuild.build_env.link_flags.append("/SAFESEH:NO /MACHINE:X64")
+                    msbuild.build(project_file="peg.sln", platforms={"x86": "Win32"}, build_type="Release", force_vcvars=True, arch="x86")
```

Comment:
https://docs.conan.io/en/latest/reference/build_helpers/visual_studio.html

> To inject the flags corresponding to the compiler.runtime, build_type and compiler.cppstd settings, this build helper also generates a properties file (in the build folder) that is passed to :command:MSBuild with :command:/p:ForceImportBeforeCppTargets="conan_build.props".

This code seems to completely ignore the build settings 😕

Reply:
Agree. @fatliverfreddy if this package would not build with other configurations, then that should be raised as ``ConanInvalidConfiguration`` in the ``validate()`` method. You can see a few cases in the repo: https://github.com/conan-io/conan-center-index/search?q=validate%28self%29&type=code

But it is not necessary to hardcode the settings, the build helper will take care of that.